client/src/app/guard/auth.guard.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@angular/core';
+import { Router, CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
+import { AuthService } from '../environments/service/auth.service';
+
+
+@Injectable({ providedIn: 'root' })
+export class AuthGuard implements CanActivate {
+    constructor(
+        private router: Router,
+        private authService: AuthService
+    ) { }
+
+    canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot) {
+        const currentUser = this.authService.currentUserValue;
+        if (currentUser) {
+            // logged in so return true
```

Comment:
I don't see a point in comments like this, it would be better if you do sth like this:
```ts
if (this.loggedIn) { return true; }

...

private get loggedIn(): boolean {
    return !!this.authService.currentUserValue;
}
```
I think it's much easier to read and doesn't leave not-really-useful comment behind :)